### ISSUE TITLE
One questions on the problem of attention model

### ISSUE THREAD

**Issue by eriche2016** [harvardnlp/seq2seq-attn#33]
Hi, Kim. Currently i am working on attention model too. But I want to add some regularization term on the attention weights to the loss function, then how to train such model model in torch?  can you give me some advice? because the attention weights are also affected by the parameters of the model, i am confused.   Very grateful if you give me some advices?

**Comment by yoonkim** [comment]
You can return the attention matrix as one of the outputs of make_lstm (through make_decoder_attn function), and then pass gradients back to that attention matrix (e.g. entropy).
